Set target temps & HVAC mode

dcrypt3d opened the pull request:
**Description:**
Add services thermostat/set_target_temps and thermostat/set_hvac_mode to component/thermostat

Tested with Ecobee which requires both target_temp_low and target_temp_high to be set at the same time... pretty sure about that.

**Related issue (if applicable):** #

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#

**Example entry for `configuration.yaml` (if applicable):**

``` yams
ecobee:
  api_key: key
  hold_temp: False

    - service: thermostat.set_hvac_mode
      data:
        entity_id: thermostat.ecobee
        hvac_mode: 'auto'

    - service: thermostat.set_target_temp_low
      data:
        entity_id: thermostat.ecobee
        target_temp_low: '65'

   - service: thermostat.set_target_temp_high
      data:
        entity_id: thermostat.ecobee
        target_temp_high: '75'

```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If code communicates with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [ ] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

balloob committed:
Merge pull request #1995 from home-assistant/dev

0.19

balloob committed:
Improve config validation error message

balloob committed:
Fix automation deprecation warning

balloob committed:
Version bump to 0.19.1

balloob committed:
Merge pull request #2004 from home-assistant/hotfix-019-1

Hotfix 019 1

turbokongen committed:
Fix for not recognizing Z-Wave thermostats (#2006)

* Fix for not recognizing thermostats

* Properly ignore zxt-120

* fix

balloob committed:
Version bump to 0.19.2

balloob committed:
Merge pull request #2010 from home-assistant/hotfix-0192

Hotfix 0192

robbiet480 committed:
en_UK->en_GB. Closes #2019.

robbiet480 committed:
Minor Fitbit tweaks. Correct the copy, dont require auth on the routes, get the client_id/client_secret from fitbit.conf instead of the YAML

robbiet480 committed:
Fitbit flake8 and pylint fixes. Forgot to do it before pushing :(

JshWright committed:
WIP: Add WSGI stack

This is a fair chunk of the way towards adding a WSGI compatible stack
for Home Assistant. The majot missing piece is auth/sessions. I was
undecided on implementing the current auth mechanism, or adding a new
mechanism (likely based on Werkzeug's signed cookies).

Plenty of TODOs...

JshWright committed:
Add some missing view registrations

Landrash committed:
Fixed minor miss-spelling (#2028)

Changed millileters to milliliters.
Changed case of mmol/l to mmol/L.

balloob committed:
Update README.rst

robbiet480 committed:
Increment version

robbiet480 committed:
Merge pull request #2033 from home-assistant/hotfix-0193

Hotfix 0193

balloob committed:
Make AND and OR conditions valid

balloob committed:
Version bump to 0.19.4

balloob committed:
Merge pull request #2038 from home-assistant/hotfix-19-4

Hotfix 0.19.4: Fix script syntax validation of AND and OR condition

balloob committed:
Fix import issues

balloob committed:
Tons of fixes - WIP

JshWright committed:
Improve view registration comments

Clarify that HomeAssistantWSGI.register_view() can handle either instantiated or uninstantiated view classes.

balloob committed:
More WIP

JshWright committed:
Friendlier exceptions for misconfigured views

If a view is missing a url or name attribute, this will result
in a more actionable exception being raised.

balloob committed:
Fix camera

balloob committed:
Add url validators

balloob committed:
Fix auth frontend

balloob committed:
Generate gzip for frontend/mdi

balloob committed:
Cache files in static folder for a year

balloob committed:
Access camera images using access token

balloob committed:
Get EventStream working

balloob committed:
Update requirements with new static update

balloob committed:
Cleanup

balloob committed:
Make event stream tests work on Travis ?

dcrypt3d committed:
Update __init__.py

dcrypt3d committed:
Update __init__.py

dcrypt3d committed:
Update ecobee.py

dcrypt3d committed:
Update ecobee.py

dcrypt3d committed:
Update services.yaml

dcrypt3d committed:
Update services.yaml

dcrypt3d committed:
Update __init__.py

dcrypt3d committed:
Update __init__.py

balloob committed:
Clean up EventStream

balloob committed:
Comment out eventstream tests

dcrypt3d committed:
Update ecobee.py

dcrypt3d committed:
Update ecobee.py

dcrypt3d committed:
Update __init__.py

dcrypt3d committed:
Update __init__.py

dcrypt3d committed:
Update services.yaml

dcrypt3d committed:
Update services.yaml

balloob committed:
Update frontend repo

balloob committed:
Add gzip for static resources

balloob committed:
Gzip all the things

balloob commented:
Instead of adding two new services, I suggest you add this functionality to the existing `set_temperature` service by adding a `temperature_range` which is a list/tuple with 2 entries.

balloob commented:
For the new service (hvac mode) and temp attributes, please add them to the demo platform and include tests.

Commit:
Better handle exceptions from Sonos players (#2085)

Sonos players can be dynamically set in various modes, for example
as TV players or Line-IN or straming from radios channels, therefore
some methods could not be available, and when invoked they cause
long exceptions to be logged. This partially solves the problem
reducing the output and logging some more informative error message

MartinHjelmare commented:
What's the difference between fan mode and hvac mode? If it has an hvac mode, maybe it should be an hvac component?

robbiet480 committed:
Accept human readable color names to change light colors (#2075)

* Add support for providing color_name which accepts a CSS3 valid, human readable string such as red or blue

* Forgot the schema validation!

* ugh farcy

* use html5_parse_legacy_color for more input options

* Add webcolors==1.5 to setup.py

* Block pylint no-member errors on tuple

* add color_name_to_rgb test

* whoops

* revert changes to individual platforms

* If color_name is set, pop it off params and set rgb_color with it

* Forgot to reset wink.py

* Import the legacy function as color_name_to_rgb directly

* reset test_color.py

* Improve light services.yaml

devdelay commented:
@balloob I like the idea of `temperature_range` since set_temperature in thermostat/ecobee.py takes `temperature` and sets it +/- 1. Ecobee's default settings are a temperature range of 5 degrees or more. I'll have to ponder how to exactly do that and wait for @nkgilley 's input/thoughts on it.

@MartinHjelmare fan mode set the hvac fan on/off that the thermostat is connected to. hvac mode is sets the mode heat, cool, auto, off etc. of the hvac unit the thermostat is connected to.

nkgilley commented:
@devdelay I've been waiting for a range slider in the frontend to implement this.  I'd agree with the idea to add the functionality to set_temperature.  When I created this component I don't think I realized that the range had to be at least 5 degrees.  When we have the range slider I'm sure we can use javascript to enforce the 5 degree range.  For now, maybe take the target_temp and ±3 to get target_low and target_high?  I don't love that idea but not sure what works better.

devdelay commented:
target_temp_low and target_temp_high are the same as set_temperature when using heating or cooling modes. When using auto mode target_temp_low sets the heating temperature and target_temp_high sets the cooling temperature. So, now I'm not sure how to proceed... both target temps have to be set at the same time. I don't really like the idea of hardcoding it to set one target temp and the other be +/- 5 when technically you could set the targets greater than 5 degrees.

nkgilley commented:
ah, yes that's right.  It's been quite some time since I've looked at any of this ecobee stuff.  I added the thermostat awhile back and honestly I barely use it in Home Assistant.  I'll keep thinking about this and let you know if I have any ideas.

happyleavesaoc committed:
add some include_dir options (#2074)

* add some include_dir options

* validate, and extend instead of add

* add yaml include tests

froz committed:
Restored telnet as an option. Activate with config option 'protocol: telnet'. Default is ssh (#2096)

robbiet480 committed:
Catch a gntp networkerror (#2099)

robbiet480 committed:
Fix profile usage with aws notify platforms (#2100)

fabaff committed:
Upgrade psutil to 4.2.0 (#2101)

fabaff committed:
Upgrade python-telegram-bot to 4.1.1 (#2102)

Danielhiversen committed:
Fix bug in flaky rfxtrx test (#2107)

pavoni committed:
Handle region enter/leave with spaces.

pavoni committed:
Ignore acc: 0 updates.

balloob committed:
Merge pull request #2108 from home-assistant/owntracks_fixes

Owntracks fixes

pavoni committed:
Bump loop energy library version.

pavoni committed:
Merge pull request #2110 from home-assistant/bump_loop_energy

Bump loop energy library version.

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Conflicts:
	homeassistant/const.py

persandstrom committed:
redirect daemon file descriptors (#2103)

jaharkes committed:
Allow for restart without using parent/child processes. (#1793)

* Allow for restart without using parent/child processes.

Assuming that we normally correctly shut down running threads and
release resources, we just do some minimal scrubbing of open file
descriptors and child processes which would stay around across an
exec() boundary.

* Use sys.executable instead of multiprocessing.spawn.get_executable()

* Limit how many file descriptors we try to close.

Don't even try to close on OSX/Darwin until we figure out how to
recognize guarded fds because the kernel will yell at us, and kill
the process.

* Use the close on exec flag on MacOS to clean up.

* Introduce a small process runner to handle restart on windows.

* Handle missing signal.SIGHUP on Windows.

wokar committed:
Added the lg_netcast platform to control a LG Smart TV running NetCast 3.0 or 4.0 (#2081)

* Added the `lgtv` platform to control a LG Smart TV running NetCast 3.0
(LG Smart TV models released in 2012) and NetCast 4.0 (LG Smart TV models released in 2013).

* Fixed multi-line docstring closing quotes

* Rename lgtv to lg_netcast

* Rename lgtv to lg_netcast

* Extracted class to control the LG TV into a separate Python package 'pylgnetcast' and changed requirements accordingly.

* regenerated requirements_all.txt with script

* now uses pylgnetcast v0.2.0 which uses the requests package for the communication with the TV

* fixed lint error: Catching too general exception Exception

turbokongen committed:
Fix fitbit KeyError (#2077)

* Fix fitbit KeyError

* Set units compared to temperature_unit

* Pass true or false for is_metric

Commit:
Add add_uri_to_queue support to (sonos) media player (#1946)

Sonos (SoCo) supports add_uri_to_queue capability, making it possible
to stream media available via HTTP for example. This patch extends
media_player component and sonos platform to support this feature

jaharkes committed:
Not sure why, but this fixed a bad filedescriptor error. (#2116)

Commit:
Add media_player.sonos_group_players service (#2087)

Sonos platform supports a `party mode` feature that groups all
available players into a single group, of which the calling player
will be the coordinator.

jaharkes committed:
Single process restart fixes (#2118)

* Ignore permission errors on setpgid.

When launched in a docker container we got a permission denied error
from setpgid.

* Don't fail if we find our own pidfile.

When we restart using exec we are running a new instance of home-assistant with
the same process id so we shouldn't be surprised to find an existing pidfile in
that case.

* Allow restart to work when started as python -m homeassistant.

When we are started with `python -m homeassistant`, the restart command line
becomes `python /path/to/hass/homeassistant/__main__.py`. But in that case the
python path includes `/path/to/hass/homeassistant` instead of `/path/to/hass`
and we fail on the first import.

Fix this by recognizing `/__main__.py` as part of the first argument and
injecting the proper path as PYTHONPATH environment before we start the new
home-assistant instance.

fabaff committed:
Docs (#2124)

* Add link to docs

* Update link

Commit:
Moldindicator Sensor (#1575)

* Adds MoldIndicator sensor platform

This sensor may be used to get an indication for possible mold growth in rooms.
It calculates the humidity at a pre-calibrated indoor point (wall, window).

* Automatic conversion to Fahrenheit for mold_indicator

* Minor change to critical temp label

* Fixed docstrings and styles

* Minor changes to MoldIndicator implementation

* Added first (non-working) implementation for mold_indicator test

* Small style changes

* Minor improvements to mold_indicator

* Completed unit test for mold indicator

* Fix to moldindicator initialization

* Adds missing period. Now that really matters..

* Adds test for sensor_changed function

robbiet480 committed:
Fix docstring issues with MoldIndicator

especially-relative committed:
Onkyo updates (#2084)

* use sane defaults for openzwave config

Use sane default if libopenzwave is installed. In most cases this will
mean that the zwave config path will not need to e manually specified.

* Resuming work on onkyo component

* Source control added to UI for onkyo receiver

Source will now display in the UI. Source mappings can be defined in the
config, and a rudimentary mapping is defined by default as a fallback.
When the onkyo source is updated, it will resolve to a defined name if
possible. This may break existing automations.

* fix lint errors

* Updated Onkyo receiver

Now takes an optional ip/name in additional to atempting to discover
deivces.

Source select will now take a sources mapping in the config. It will
provide default values if no source mapping is provided.

example:

- platform: onkyo
  host: 10.0.0.2
  name: receiver
  sources:
    HTPC: 'pc'
    Chromecast: 'aux1'
    Bluray: 'bd'
    Wii U: 'game'

* fix pylint error

* Use HA's error log instead of stack trace

* Flipped source mappings, code cleanup

kellerza committed:
Updated pyqwikswitch & QS<->HA UI behaviour (#2123)

* Updated pyqwikswitch & constants

* Disable too-many-locals

nkgilley committed:
Change color_RGB_to_xy formula & return brightness (#2095)

* Use RGB to XY calculations from Philips Hue developer site

* uppercase X,Y,Z

* rename cx,cy to x,y

* return brightness in color_RGB_to_xy

* remove try/catch

* update existing platforms using color_RGB_to_xy

* improve wemo w/ jaharkes suggestion

* allow brightness override of rgb_to_xy

ishults committed:
#2120 Fix hvac z-wave fan list (#2121)

* #2120 Fix hvac z-wave fan list

* Properly name methods

robbiet480 committed:
GTFS fixes (#2119)

* Change to official PyGTFS source

* Threading fixes for GTFS

* Actually pygtfs 0.1.3

* Update requirements_all.txt

* Update gtfs version

omgapuppy committed:
Add support for Logentries (#1945)

* Add support for Logentries

Supports sending has events to Logentries web hook endpoint
see logentries.com for more

Inspired by the Splunk component

* bugfix

* fix summary

* fix test

* fix logentries url and tests

* update tests

* mock token

* Bug fixes

* typo

* typo

* fix string splitting

* remove redundant backslash

balloob committed:
Upgrade Nest to 2.9.2 (#2126)

jaharkes committed:
Let systemd handle home-assistant process restarts. (#2127)

balloob committed:
Update frontend

balloob committed:
Merge pull request #2113 from home-assistant/dev

0.20

balloob committed:
Version bump to 0.20

balloob committed:
Version bump to 0.21.0.dev0

balloob committed:
Remove gzip API

balloob committed:
Remove unused import

balloob committed:
Merge pull request #2063 from home-assistant/feature/wsgi

Feature/wsgi

balloob committed:
update frontend gz

balloob committed:
Another attempt to fix SSL in Docker

balloob committed:
EventStream to sent ping on start to notify browser

balloob committed:
Fix Dockerfile

balloob committed:
Use only 1 event listener for event stream.

jaharkes committed:
Ignore assertions from python threading when looking for leaked threads.

While looking for leaked resources (threads) after shutdown and before restart
we in some cases get an assertion in the python threading module where we find
a thread marked as running at the python level but it has no associated thread
at the C level.

jaharkes committed:
Version bump to 0.20.1

jaharkes committed:
Ignore assertions from python threading when looking for leaked threads. (#2130)

While looking for leaked resources (threads) after shutdown and before restart
we in some cases get an assertion in the python threading module where we find
a thread marked as running at the python level but it has no associated thread
at the C level.

balloob committed:
Merge pull request #2132 from home-assistant/hotfix-20-1

Hotfix 0.20.1

partofthething committed:
Added LIRC component for responding to IR remote commands

partofthething committed:
added LIRC component to .coveragerc

partofthething committed:
Removed LIRC dependency from requirements due to "complex" compliation

User will have to install lirc and python-lirc manually.

balloob committed:
Upgrade eventlet to 0.19

partofthething committed:
LIRC: Responded to some code review requests but not the big one

robbiet480 committed:
Update issue template to prettify the header.

robbiet480 committed:
If no departure time is set, use now as the default. If departure time is set but does not have a :, assume its a preformed Unix timestamp and send along as raw input. Assume same for arrival_time.

robbiet480 committed:
If no departure time is set, use now as the default. If departure time is set but does not have a :, assume its a preformed Unix timestamp and send along as raw input. Assume same for arrival_time.

robbiet480 committed:
If we have duration_in_traffic use that as the state, otherwise use duration

robbiet480 committed:
If we have duration_in_traffic use that as the state, otherwise use duration

JshWright committed:
Fix TLS with eventlet (#2151)

* Fix TLS with eventlet

This fixes a simple error on my part when implementing the WSGI stuff.

eventlet.wrap_ssl() returns a wrapped socket, it does not modify the
object passed to it. We need to grab the returned value and use that.

* Fix style issue

jaharkes committed:
Don't even bother trying to kill stray child processes.

When we change our process group id we don't get keyboard interrupt
signals passed if our parent is a bash script.

jaharkes committed:
Don't even bother trying to kill stray child processes.

When we change our process group id we don't get keyboard interrupt
signals passed if our parent is a bash script.

jaharkes committed:
Version bump to 0.20.2

partofthething committed:
Changed LIRC component so that it just fires events on the bus.

partofthething committed:
Moved lirc out of sensor package.

partofthething committed:
Updated requirements for LIRC

balloob committed:
Merge pull request #2154 from home-assistant/hotfix-20-2

Hotfix 20 2

balloob committed:
manifest.json: remove trailing commas

wokar committed:
lg_netcast: fix exception on missing access_token (#2150)

* specified default value for acccess_token to prevent exception on init

balloob committed:
WSGI: Hide password in logs (#2164)

* WSGI: Hide password in logs

* Add auth + pw in logs tests

sxb1n9 committed:
Update PyISY to 1.0.6 (#2133)

* Update PyISY to 1.0.6 

1.0.6 is the newest version of PyISY

* PyISY to 1.0.6

fabaff committed:
Upgrade astral to 1.1 (#2131)

Commit:
Round temp and percentage for octoprint sensors (#2128)

partofthething committed:
Merge remote-tracking branch 'origin/dev' into lirc

partofthething committed:
Cleanups to LIRC module

Danielhiversen committed:
Fix bug in google travel time. Default option dictionary must contain mode. (#2134)

Commit:
Fixed issue with edimax SP-1101 switches (#2105)

* Fixed issue with edimax SP-1101 switches

* Added missing ValueError exception

partofthething committed:
Locked in required version of python-lirc for LIRC component

balloob committed:
Handle invalid dev ids for dev tracker + owntracks (#2174)

balloob committed:
Update frontend gzip

balloob committed:
Update frontend

balloob committed:
Update frontend

balloob committed:
Bugfixes for urls with dates

balloob committed:
Fix lint errors

balloob committed:
Update frontend

balloob committed:
Update frontend

rbray89 commented:
I'd love to have this as well, my new Z-wave thermostat allows for a Cool and Heat temp.
One thing I'd like is a min-range, as my thermostat must have 3 degrees between the two temps.

balloob committed:
Update frontend

Commit:
Add Vagrant setup (#2171)

balloob committed:
Update frontend

fabaff committed:
Vendorize vincenty requirement (#2176)

Commit:
Fixed roku exception when device is powered off or looses connection (#2173)

rubund committed:
New option for the netatmo platform: station (#2178)

This is necessary if multiple weather stations are associated with
one Netatmo account.

rubund committed:
Initial support for EnOcean (#2177)

* Initial support for EnOcean

Tested to work with:
 - Eltako FUD61 dimmer
 - Eltako FT55 battery-less switch
 - Permundo PSC234 (switch and power monitor)

* Rerun gen_requirements_all.py

olimpiurob committed:
Added Osram lightify platform (#2170)

* Added Osram Lightify light component

* Added color temperature and fade transition support to Osram Lightify

* Added Osram Lightify light component

* Added color temperature and fade transition support to Osram Lightify

* Updated docstring

* Added osramlightify to ignore list on coveragerc and updated docstrings

* Fixed linting issues

wkonkel committed:
Insteon support for brightness (#2169)

* Insteon support for brightness

* Farcy fix for unused constants.

* Remove unused constant and fix whitespace.

* Prevent toggle switches from jumping between states.

* 255 not 256

jaharkes committed:
Allow time condition windows to cross midnight. (#2158)

* Allow time condition windows to cross midnight.

* Address comments.

Fold _in_time_window back into the time() condition test.
Use specific time values to test the time window.

Commit:
Feedreader: add file data storage (#2147)

Right now we ignore already parsed entries and store the information
at runtime, but it will not survive a restart. This patch adds storage
functionality storing pickled file into default config folder when
feed has `published_parsed` support.

Bart274 committed:
List entity_ids in config and only react to them (#2144)

* List entity_ids in config and only react to them

This allows us to define a list of entity_ids in the config to make the
template sensor, binary sensor and switch only react to state changes of
these entities instead of listening to all state changes.

* Forgot to import the track_state_change function

* Changed test for added entity_ids to config

* Use default MATCH_ALL and remove event_listener

balloob committed:
Merge branch 'pr/2139' into dev

Conflicts:
	.coveragerc

fabaff committed:
Add azimuth (#1951)

* Add azimuth

* Place elevation and azimuth together in update part

danieljkemp commented:
Maybe splitting this into two PRs would be a good idea? I plan to add the HVAC mode control to radiothermostat as soon as this goes in.

balloob committed:
Fix linting issues

balloob committed:
Fix broken tests + linting

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

Conflicts:
	homeassistant/const.py

balloob committed:
Improve index.html template

themactep committed:
Improve Internet Time calculation. (#2185)

JshWright committed:
Re-add config validation for the http component (#2186)

This commit adds back the config validation for the http component. It
was removed during the WSGI shuffle. This is just a direct copy of what
@robbiet480 added in ab294d12f7 (with some testing to verify it still
works).

fabaff committed:
Remove print (already covered by logger) (#2184)

balloob committed:
Optimize foreacast.io API calls

fabaff committed:
Add windows 10 tile (#2166)

balloob committed:
Merge pull request #2191 from home-assistant/forecast-api-calls

Optimize foreacast.io API calls

robbiet480 committed:
Report whether SSL is required in discoverables, like /api/discovery_info and ZeroConf (#2192)

robbiet480 committed:
Revert "Report whether SSL is required in discovery"

balloob committed:
Merge pull request #2196 from home-assistant/revert-2192-expose-required-ssl-in-discoveries

Revert "Report whether SSL is required in discovery"

balloob committed:
Fix Norway (#2197)

nkgilley committed:
flux platform as a switch (#2097)

* flux platform as a switch

* use track_time_change. broken :(

* use track_utc_time_change instead of track_time_change

* add some basic tests

* use brightness from RGB_to_xy

* config_schema validation

* back to platform schema. what was i doing?

* more broken tests :(

* 644

* fix some time bugs

* add working tests. config validation still not right

* bug fixes and more test cases.

balloob committed:
Update frontend

infamy committed:
Update PULL_REQUEST_TEMPLATE.md (#2198)

* Update PULL_REQUEST_TEMPLATE.md

* Update PULL_REQUEST_TEMPLATE.md

jacobtomlinson committed:
Added rfxtrx rollershutter (#2030)

* Added rfxtrx rollershutter

* Updated mock command with real one

* Corrected test string

pavoni committed:
Bump pywemo version to fix ssdp discovery encoding issue.

pavoni committed:
Merge pull request #2201 from home-assistant/fix_pywemo_ssdp_decode_utf8

Bump pywemo version to fix ssdp discovery encoding issue.

pavoni committed:
Bump loopenergy version. Increased interval before deciding connection is dead and reconnecting.

pavoni committed:
Merge pull request #2203 from home-assistant/bump_loopenergy

Bump pyloopenergy version.

samriley committed:
Support for RFY protocol (#2199)

fabaff committed:
Upgrade python-telegram-bot to 4.2.0 (#2204)

kellerza committed:
Update Qwikswitch library version (#2214)

dcrypt3d committed:
myq

dcrypt3d committed:
myq

dcrypt3d committed:
Merge branch 'target-hvac' of https://github.com/devdelay/home-assistant into target-hvac

dcrypt3d committed:
update

balloob commented:
Could you clean up your branch? It currently has 167 changed files…

devdelay commented:
Rebased on that branch when pulling down .22dev... whoops. I was thinking I would create a new PR later that only adds set_hvac_mode service. I just noticed it was added to ecobee.py but the service is not in __init__.py
